fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordValidationTest.java

```diff
@@ -0,0 +1,238 @@
+        // Delete the versions
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION);
+            Tuple pkAndVersion = result.get(0).getPrimaryKey().add(-1);
+            byte[] versionKey = store.recordsSubspace().pack(pkAndVersion);
+            store.ensureContextActive().clear(versionKey);
+
+            pkAndVersion = result.get(1).getPrimaryKey().add(-1);
+            versionKey = store.recordsSubspace().pack(pkAndVersion);
+            store.ensureContextActive().clear(versionKey);
+            commit(context);
+        }
+
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION);
+            validateRecord(store, result.get(0), FDBRecordStoreBase.RecordValidationOptions.VALID_VERSION);
+            validateRecord(store, result.get(1), FDBRecordStoreBase.RecordValidationOptions.VALID_VERSION);
+            commit(context);
+        }
+    }
+
+    @Nonnull
+    private static RecordMetaDataHook getRecordMetaDataHook(final boolean splitLongRecords) {
+        return metaData -> {
+            metaData.setSplitLongRecords(splitLongRecords);
+            // index cannot be used with large fields
+            metaData.removeIndex("MySimpleRecord$str_value_indexed");
+        };
+    }
+
+    private void validateRecord(FDBRecordStore store, final FDBStoredRecord<Message> rec, final FDBRecordStoreBase.RecordValidationOptions validationOption) throws Exception {
+        EnumSet<FDBRecordStoreBase.RecordValidationOptions> recordValidationOptions;
+        recordValidationOptions = store.validateRecordAsync(rec.getPrimaryKey(), EnumSet.allOf(FDBRecordStoreBase.RecordValidationOptions.class), false).get();
+        assertNotNull(recordValidationOptions);
+        EnumSet<FDBRecordStoreBase.RecordValidationOptions> expected = (validationOption == null) ?
+                                                                       EnumSet.noneOf(FDBRecordStoreBase.RecordValidationOptions.class) :
+                                                                       EnumSet.of(validationOption);
+        assertEquals(expected, recordValidationOptions);
+
+        recordValidationOptions = store.validateRecordAsync(rec.getPrimaryKey(), EnumSet.noneOf(FDBRecordStoreBase.RecordValidationOptions.class), false).get();
+        assertNotNull(recordValidationOptions);
+        assertEquals(EnumSet.noneOf(FDBRecordStoreBase.RecordValidationOptions.class), recordValidationOptions);
+    }
+
+
+    @Nonnull
+    private List<FDBStoredRecord<Message>> saveRecords(final boolean splitLongRecords, final int formatVersion, final RecordMetaDataHook hook) throws Exception {
```

Comment:
It seems like it would be clearer to have two methods, and use those rather than a method that returns a list, namely `saveUnsplitRecord` and `saveSplitRecord`

Reply:
The returned records are always a list of 2. The difference is that sometimes one of the records has a long text that gets split across 3 splits. When long records are not supported, both records will be short.